include/swift/AST/PrintOptions.h

```diff
@@ -308,6 +308,12 @@ struct PrintOptions {
   /// \brief The information for converting archetypes to specialized types.
   std::shared_ptr<ArchetypeTransformContext> TransformContext;
 
+  /// \brief If this is not \c nullptr then functions (incuding accessors and
+  /// constructors) will be printed with a body that is determined by this
+  /// function.
+  std::function<std::string(const ValueDecl *)> *FunctionBody =
```

Comment:
std::function has a default-constructed null state; is there a reason why you aren't using it?